blog/_posts/2021-02-16-improving-scala-library-ecosystem.md

```diff
@@ -0,0 +1,127 @@
+---
+layout: blog-detail
+post-type: blog
+by: Eugene Yokota, Julien Richard-Foy
+title: Improving the Scala Library Ecosystem
+---
+
+One of the things that makes Scala powerful and fun to use is its library ecosystem — spanning across the Maven ecosystem for JVM, and npm for Scala.JS. We can use libraries to do operations like download files from the web, or we can use it to adopt different programming paradigms. The library ecosystem allows us to write more useful programs with less effort.
+
+## What is a version conflict?
+
+However, the library ecosystem is not without problems. A library that you pulled could depend on other libraries, and the transitive dependencies could cause version conflicts. Here's a quick example of a Scala project that uses Akka HTTP, a Postgres database, and JSON. Its build declares two library dependencies, `akka-http-circe` and `doobie-postgres-circe`:
```

Comment:
I wonder if it's worth explicitly this as a general JVM problem, not a Scala-specific problem. As JVM vets that's ingrained in us, but not all readers are JVM 4 Life.

Reply:
Also, some readers might like to know if the _solution_ only works for Scala libraries, or whether Java libraries could use it too.

Reply:
While you're right that this stems from the loose behavior of the dependency resolvers used in JVM, given we have Scala.js and Native, I don't think we should make it too JVM centric either.